dashboard/db/migrate/20180110233855_create_seeded_s3_objects.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSeededS3Objects < ActiveRecord::Migration[5.0]
+  def change
+    create_table :seeded_s3_objects do |t|
+      t.string :bucket
+      t.string :key
+      t.string :etag
+
+      t.timestamps
+    end
+
+    add_index :seeded_s3_objects, [:bucket, :key, :etag]
```

Comment:
Should we also have an index on just `[:bucket, :key]` without `:etag`? I can see wanting a quick check to know whether a particular S3 file is seeded at all (regardless of contents / version).

Reply:
We shouldn't need an additional index. Since bucket and key are the leading edge of the index it should be used for queries even if they don't include an etag. 

```
mysql> explain select * from seeded_s3_objects o where o.bucket='a bucket' and o.key='a key';
explain select * from seeded_s3_objects o where o.bucket='a bucket' and o.key='a key';
+----+-------------+-------+------------+------+----------------------------------------------------+----------------------------------------------------+---------+-------------+------+----------+-------+
| id | select_type | table | partitions | type | possible_keys                                      | key                                                | key_len | ref         | rows | filtered | Extra |
+----+-------------+-------+------------+------+----------------------------------------------------+----------------------------------------------------+---------+-------------+------+----------+-------+
|  1 | SIMPLE      | o     | NULL       | ref  | index_seeded_s3_objects_on_bucket_and_key_and_etag | index_seeded_s3_objects_on_bucket_and_key_and_etag | 1536    | const,const |    1 |   100.00 | NULL  |
+----+-------------+-------+------------+------+----------------------------------------------------+----------------------------------------------------+---------+-------------+------+----------+-------+
1 row in set, 1 warning (0.01 sec)
```